### PR TITLE
Report docstring content violations on docstring line (fixes #83)

### LINKED ISSUE #83: D401 reports incorrect line number
When reporting a D401, the line number returned is the line of the function, not the line number for the docstring

### PR DESCRIPTION
**Pull request by lordmauve**
Add `.start` and `.end` attributes to docstring objects that hold the starting and ending line numbers of the docstring token. Docstring was changed to a subclass of str to allow this, ensuring backwards compatibility.

Add a property `.error_lineno` to Definition. This will use the docstring's line number if it exists, or the definition's line number otherwise.

Finally, switch `Error` to use the new `.error_lineno` property.

Tested on Python 2.7, 3.3-3.5 and PyPy with Tox.

### CODE REVIEW

File: src/pydocstyle/parser.py [hunk 1]
```diff
@@ -189,6 +196,21 @@ class Decorator(Value):
...
 
 
+class Docstring(str):
+    """Represent a docstring.
+
+    This is a string, but has additional start/end attributes representing
+    the start and end of the token.
+
+    """
+    def __new__(cls, v, start, end):
```
Comment (Nurdok) [inline src/pydocstyle/parser.py]: Why is this needed? Why not just call `super` in `__init__`?

Comment (lordmauve) [inline src/pydocstyle/parser.py]: IIRC, you can't subclass various native types without defining __new__().

Comment (lordmauve) [inline src/pydocstyle/parser.py]: I confirmed, this is needed.

**Comment by lordmauve** [comment]
Bump. Is merging this blocked by #240?

**Comment by lordmauve** [comment]
Disappointed this didn't make 2.0.0. Please let me know what needs to happen for this to be merged.

**Comment by Nurdok** [comment]
@lordmauve sorry about that. I think this will go in the next minor version.

**Comment by lordmauve** [comment]
I just rebased. @Nurdok, can this be merged?

**Comment by Nurdok** [comment]
Hi @lordmauve, can you please just add a line to the release notes first?

**Comment by lordmauve** [comment]
I've added to the notes under a new minor version number. The release date will need to be added when you come to release this.

File: docs/release_notes.rst [hunk 2]
```diff
@@ -4,6 +4,14 @@ Release Notes
 **pydocstyle** version numbers follow the
 `Semantic Versioning <http://semver.org/>`_ specification.
 
+2.2.0 - not yet released
```
Comment (Nurdok) [inline docs/release_notes.rst]: Just put "Current Development Version" in the title instead of specifying the actual version.

Comment (lordmauve) [reply docs/release_notes.rst]: Done.

File: docs/release_notes.rst [hunk 3]
```diff
@@ -4,6 +4,14 @@ Release Notes
 **pydocstyle** version numbers follow the
 `Semantic Versioning <http://semver.org/>`_ specification.
 
+Current development version
+---------------------------
+
+New features
+
+* Violations are now reported on the line where the docstring starts, not the
+  line of the ``def``/``class`` it corresponds to.
```
Comment (Nurdok) [inline docs/release_notes.rst]: Link to both this (#238) and #83.

Sorry for the nitpicking, in a few moments this will be merged :)

Comment (lordmauve) [reply docs/release_notes.rst]: Done.
